clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -558,12 +558,14 @@ public void handle(JoinGroupResponse joinResponse, RequestFuture<ByteBuffer> fut
                 markCoordinatorUnknown();
                 log.debug("Attempt to join group failed due to obsolete coordinator information: {}", error.message());
                 future.raise(error);
+            } else if (error == Errors.FENCED_INSTANCE_ID) {
+                log.error("Received fatal exception: group.instance.id {} gets fenced", groupInstanceId);
+                future.raise(error);
             } else if (error == Errors.INCONSISTENT_GROUP_PROTOCOL
                     || error == Errors.INVALID_SESSION_TIMEOUT
                     || error == Errors.INVALID_GROUP_ID
                     || error == Errors.GROUP_AUTHORIZATION_FAILED
-                    || error == Errors.GROUP_MAX_SIZE_REACHED
-                    || error == Errors.FENCED_INSTANCE_ID) {
+                    || error == Errors.GROUP_MAX_SIZE_REACHED) {
```

Comment:
If we are fenced, should we keep track of that somewhere so that we do not keep sending RPCs to the coordinator?

Reply:
As long as we don't reset our generation info, all subsequent requests should be failing once other consumer joins the group right? Eventually this will lead to a complete crash IIUC.

Reply:
I think I wasn't clear. What I'm asking is whether the consumer should remember the fact that it was fenced. So if the user continues trying to do stuff, we fail immediately instead of sending additional requests to the broker.

Reply:
For `Join/Sync/OffsetCommitSync` the failures should be immediate; for `heartbeat/commitAsync` it would not be immediate. If we do want a global variable indicating the failure, potentially we need to add a new `MemberState`

Reply:
Yes, a new MemberState would be a nice way to handle this. We can do this later if you do not think it is important now.

Reply:
SG!